### ISSUE TITLE
Add MudBlazor form validation integration to MudExUploadEdit

### ISSUE THREAD

**Issue by JaronMLifsey** [fgilde/MudBlazor.Extensions#57]
### Feature request type

Enhance component

### Component name

MudExUploadEdit

### Is your feature request related to a problem?

MudExUploadEdit does not support custom validation on form submission.

### Describe the solution you'd like

Add custom validation to the components. Ideally, they would extend MudFormComponent and so integrate with other MudBlazor form components. For example, I want to add validation such that the user must upload a specific set of 3 files with specific names and form validation should fail if that's not the case when they hit submit.

### Have you seen this feature anywhere else?

MudBlazor FileUpload.

### Describe alternatives you've considered

I've created a wrapper component to add validation to arbitrary components. This works well enough, but ideally the extension components would act like MudBlazor Form components on its own. Here's the code I used if interested in implementing this request. A real solution would likely have the components inherit from MudFormComponent to integrate with mud forms and call validation whenever the user adds a file (not after the file is fully read from like this workaround).

ValidationWrapper.razor
```
@typeparam T
@inherits MudFormComponent<T, string>

<MudInputControl Class="@Class" Style="@Style" Error="HasErrors" ErrorText="@GetErrorText()" Required="@Required">
    <InputContent>
        @ChildContent
    </InputContent>
</MudInputControl>

@code {
    /// <summary>
    /// The value of the ValidationWrapper component. This will be passed to the set validation function.
    /// </summary>
    [Parameter]
    [EditorRequired]
    public T? Value
    {
        get => _value;
        set
        {
            if (_value != null && _value.Equals(value))
            {
                return;
            }
            _value = value;
        }
    }
    public ValidationWrapper() : base(new DefaultConverter<T>()) { }

    [Parameter]
    public RenderFragment? ChildContent { get; set; }
}
```

And then the usage:
```
  <ValidationWrapper @ref="FileUploadValidator"
                     T="IReadOnlyList<UploadableFile>"
                     Required="true"
                     Value="@Files"
                     Class="fileUploadContainer"
                     Validation="@FilesValidator">
      <MudExUploadEdit @ref="UploadComponent"
                       T="UploadableFile"
                       Extensions="@(new string[]{".xlsx"})"
                       TextUploadFile="Click here to upload files"
                       TextDropZone="Drag and Drop files here"
                       ColorizeIcons="true"
                       MinHeight="80"
                       AllowFolderUpload="false"
                       AllowExternalUrl="false"
                       Class="MudExUploadExCustom"
                       AllowRename="true"
                       UploadRequests="@Files"
                       UploadRequestsChanged="OnFilesChanged"
                       AllowPreview="false"></MudExUploadEdit>
  </ValidationWrapper>

@code {
    private ValidationWrapper<IReadOnlyList<UploadableFile>> FileUploadValidator = default!;
    private List<UploadableFile> Files = new();
    private Func<IReadOnlyList<UploadableFile>, string?> FilesValidator { get; private set; }
    private Task OnFilesChanged(IList<UploadableFile> files)
    {
        return FileUploadValidator.Validate();
    }
}
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by fgilde** [comment]
I will implement it

**Comment by fgilde** [comment]
Done. Will be included in release 1.7.76 but you can have a look at the preview release where its already included. [1.7.76-prev-2311021449-main](https://www.nuget.org/packages/MudBlazor.Extensions/1.7.76-prev-2311021449-main)

Please close the issue if this works for you
